### ISSUE TITLE
Question about box density/mass parameter usage in carrying task

### ISSUE THREAD

**Issue by LeyangWen** [liangpan99/TokenHSI#6]
Hi,

Thank you for the great work. 

I noticed that there is a box density parameter that can be set here:
https://github.com/liangpan99/TokenHSI/blob/c6cd679c1ec7517ecce77a49ffb269362c02385a/tokenhsi/data/cfg/basic_interaction_skills/amp_humanoid_carry.yaml#L56
https://github.com/liangpan99/TokenHSI/blob/c6cd679c1ec7517ecce77a49ffb269362c02385a/tokenhsi/env/tasks/basic_interaction_skills/humanoid_carry.py#L274

However, I couldn’t find any trained policies that made use of it. Also, in the paper’s supplementary material (end of Section A), you mentioned that `no external forces are applied to any rigid body of the simulated character`, which seems to contradict the notion of using the box mass. 

Given that, I was wondering if the density parameter (or box mass) is used or tested for the carrying task. I tried training a policy with `randomDensity: True`, but noticed that the reported reward plateaued around 190, compared to around 317 during normal training. During inference, the humanoid often gets stuck trying to lift the box or misses the box target location. Am I missing something or doing something wrong?

Thanks for your time and great work on this project!

**Comment by liangpan99** [comment]
```no external forces are applied to any rigid body of the simulated character``` means we do not let the policy output additional 6 values that are external forces applied on the simulated character's root joint. We emphasize this because there are some phyiscs-based character control papers using that setting. And we want to distinguish our paper from those papers.

And this is unrelated to if we randomize the box mass.

The ```randomDensity``` is a very old experimental config, which is already legacy in the current version. Both config and hyper-parameters for randomly sampling box densities are legacy.

Now, the current code is sampling densities from [300, 800]. This results in too many heavy boxes that the character cannot afford. I recommend to fine-tune these two parameters to make sure that the masses of all generated boxes are reasonable.

**Comment by LeyangWen** [comment]
Thank you for the fast response.

That makes sense to me. My experiments with the [80, 120] range are giving me a more similar curve to the original training.

Thank you so much for your help.
